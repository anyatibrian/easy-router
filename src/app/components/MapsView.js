import React from 'react'
import { View, StyleSheet, Text, TouchableHighlight,ActivityIndicator} from 'react-native'
import MapView,{PROVIDER_GOOGLE,Marker, Polyline} from 'react-native-maps'
import Icons from './Icons'

const MapsView = ({coordinates, handleRoutes,loading}) => {
    const coords = coordinates.map(([longitude, latitude])=>{
        return { latitude, longitude}
    })
    return (
        <View>
             <MapView
             style={styles.container}
             provider={PROVIDER_GOOGLE}
             showsUserLocation={true}
             region={{
             latitude:0.318522,
             longitude:32.586076,
             latitudeDelta:0.20,
             longitudeDelta:0.20
             }}>
             { coords.length !==0?
                <Polyline
                 coordinates={[...coords]}
                 strokeWidth={4}
                 strokeColor="black"
                 />:null
            }
            {
              coords[coords.length-1]? <Marker
              coordinate={ coords[coords.length-1]}
              /> : null
            }
            
            {
               coords[0]? <Marker
               coordinate={ coords[0]}
               pinColor="green"
               />:null
            }
            </MapView>
            <View style={styles.rtContainer}>
                <TouchableHighlight onPress={()=>handleRoutes('car')} style={styles.touchables} >
                    {loading ? <ActivityIndicator size="large" color="#ffff" style={styles.indication}/>:<Icons 
                        style={styles.Icons}
                        name="directions-car" 
                        size={40}/> 
                    }
                </TouchableHighlight>
                <TouchableHighlight style={styles.touchables} onPress={()=>handleRoutes('bike')}>
                    {loading ? <ActivityIndicator size="large" color="#ffff" style={styles.indication}/>:
                        <Icons 
                        style={styles.Icons}
                        name="motorcycle" 
                        size={40}/>
                    }
                </TouchableHighlight>
                <TouchableHighlight style={styles.touchables} onPress={()=>handleRoutes('foot')}>
                    {loading ? <ActivityIndicator size="large" color="#ffff" style={styles.indication}/>:
                        <Icons 
                        style={styles.Icons}
                        name="directions-walk" 
                        size={40}/> 
                    }
                </TouchableHighlight>
            </View>
        </View>
    )
}

export default MapsView

const styles = StyleSheet.create({
    container: {
        height:'100%',
        width:'100%',
        zIndex:-1
      },
      map: {
         ...StyleSheet.absoluteFillObject
      },
      rtContainer:{
          flexDirection:"row",
          marginTop:'130%',
          position:'absolute',
          margin:40
      },
      Icons:{
          color:'white',
          padding:20
      },
      touchables:{
          display:'flex',
          borderColor:'#f57b51',
          backgroundColor:'#f57b51',
          height:80,
          width: 80,
          marginLeft:20,
          borderRadius:100
      },
      indication:{
          padding:20
      }
})
