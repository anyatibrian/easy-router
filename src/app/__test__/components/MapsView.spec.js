import React from 'react'
import { shallow } from 'enzyme'
import MapsView from '../../components/MapsView'

const props={
    coordinates:[], 
    handleRoutes:jest.fn(),
    loading:false,
    destination:'kampala',
    origin:'lira'
}
let wrapper
let instance
describe('MapsView component', () => {
    
    beforeEach(()=>{
        wrapper = shallow(<MapsView {...props}/>)
        instance = wrapper.instance()
    })

    it("should test the component renders without fail", ()=>{
        expect(wrapper).toMatchSnapshot()
    })
    it('it should test is loading state', ()=>{
        wrapper.setProps({loading:true})
        expect(props.loading).toEqual(false)
    }),
    it('should test the poyline displayed on the map', ()=>{
        wrapper.setProps({coordinates: [[12457,67848]]})
        wrapper.find('Polyline')
        expect(wrapper).toMatchSnapshot()
    }),
    it(' should test the onpress method', ()=>{
       const onPress0 =  wrapper.find('TouchableHighlight').at(0)
       const onPress1 =  wrapper.find('TouchableHighlight').at(1)
       const onPress2 =  wrapper.find('TouchableHighlight').at(2)
       onPress0.props().onPress()
       onPress1.props().onPress()
       onPress2.props().onPress()
       expect(props.handleRoutes).toHaveBeenCalled()
    }),
    it('should test the is loading state', ()=>{
        wrapper.setProps({loading:true})
        wrapper.find('ActivityIndicator').at(0)
        wrapper.find('ActivityIndicator').at(1)
        wrapper.find('ActivityIndicator').at(2)
        wrapper.setProps({loading:false})
        wrapper.find('Icons').at(0)
        expect(wrapper).toMatchSnapshot()
    })
})
