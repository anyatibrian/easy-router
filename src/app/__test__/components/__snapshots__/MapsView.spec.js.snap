// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapsView component should test the component renders without fail 1`] = `ShallowWrapper {}`;

exports[`MapsView component should test the is loading state 1`] = `ShallowWrapper {}`;

exports[`MapsView component should test the poyline displayed on the map 1`] = `ShallowWrapper {}`;
